pkg/solana/chainwriter/chain_writer.go

```diff
@@ -316,10 +390,41 @@ func (s *SolanaChainWriterService) SubmitTransaction(ctx context.Context, contra
 	discriminator := GetDiscriminator(methodConfig.ChainSpecificName)
 	encodedPayload = append(discriminator[:], encodedPayload...)
 
+	// Fetch latest blockhash
+	blockhash, err := s.client.LatestBlockhash(ctx)
+	if err != nil {
+		return errorWithDebugID(fmt.Errorf("error fetching latest blockhash: %w", err), debugID)
+	}
+
+	if len(createATAinstructions) > 0 {
+		ataTx, ataErr := solana.NewTransaction(
+			createATAinstructions,
+			blockhash.Value.Blockhash,
+			solana.TransactionPayer(feePayer),
+		)
+		if ataErr != nil {
+			return errorWithDebugID(fmt.Errorf("error constructing ATA transaction: %w", err), debugID)
+		}
+
+		// Enqueue ATA transaction
+		if err = s.txm.Enqueue(ctx, methodConfig.FromAddress, ataTx, &transactionID, blockhash.Value.LastValidBlockHeight); err != nil {
```

Comment:
I think we should create a custom transactionID for the ATA transactions. Maybe something like `ATA-{uuid}`? Otherwise, the TXM will have a collide on this ID.

Reply:
Good catch!
If we reuse the transactionID, the TXM will always return ErrIDAlreadyExists when you actually send the main Tx.
I would recommend creating a new uuid here itself, for the ATA tx. This will allow us to track its status.

Using ATA-{uuid} can be problem if the caller hadn't set uuid. Because then we'd use "ATA-" as the id for all ATA txs, and they will then collide with each other.

Reply:
Oh sorry I sort of wrote that confusingly. I meant we can just generate a uuid on the spot since this txID is transient. We only need it for the next step when we poll the status so something like `ATA-{uuid.NewString()}` is what I meant

Reply:
added a custom txID in the form `fmt.Sprintf("ATA-%s", uuid.NewString())`